src/Sylius/Bundle/ProductBundle/Form/Type/ProductOptionValueCollectionType.php

```diff
@@ -31,14 +32,15 @@
 {
     /**
      * {@inheritdoc}
+     *
+     * @throws \InvalidArgumentException
      */
-    public function buildForm(FormBuilderInterface $builder, array $options)
+    public function buildForm(FormBuilderInterface $builder, array $options): void
     {
-        if ($this->areOptionsValid($options)) {
-            throw new InvalidConfigurationException(
-                'array or (\Traversable and \ArrayAccess) of "Sylius\Component\Variation\Model\OptionInterface" must be passed to collection'
-            );
-        }
+        Assert::false(
+            $this->areOptionsValid($options),
```

Comment:
Side note: I'd move throwing the exception to that method, making it `assertOptionsAreValid` instead.